physical/tvf.go

```diff
@@ -187,6 +187,35 @@ func (node *TableValuedFunction) Materialize(ctx context.Context, matCtx *Materi
 
 		return tvf.NewTumble(matSource, timeField, matWindowLength, matWindowOffset), nil
 
+	case "zero":
+		source, err := node.getArgumentTable(octosql.NewVariableName("source"))
+		if err != nil {
+			return nil, err
+		}
+
+		matSource, err := source.Materialize(ctx, matCtx)
+		if err != nil {
+			return nil, errors.Errorf("couldn't materialize source")
+		}
+
+		return tvf.NewZeroWatermark(matSource), nil
+
+	case "watermark":
+		source, err := node.getArgumentTable(octosql.NewVariableName("source"))
+		if err != nil {
+			return nil, err
+		}
+		timeField, err := node.getArgumentDescriptor(octosql.NewVariableName("time_field"))
```

Comment:
You can get the timefield from a record. There's a function for that.

Reply:
yea but that's the way to do this in tumble/range so i copied it

Reply:
what about this:

`timeField := source.Metadata().EventTimeField()`

and keep the tvf/watermark file unchanged?

Reply:
Tumble is more general. This could be an optional argument.

Yes, your proposal looks good.